Storage/Processor/Processor.php

```diff
@@ -259,4 +248,72 @@ public function getResource(Config $config, string $filename, bool $noCache = fa
         $this->saveGeneratedResourceToCache($generatedResource, $config, $filename);
         return $generatedResource;
     }
+
+    private function getResponseFromStreamInterface(StreamInterface $streamInterface, Request $request): StreamedResponse
+    {
+        $response = new StreamedResponse(function () use ($streamInterface) {
+            if ($streamInterface->isSeekable()) {
+                $streamInterface->rewind();
+            }
+
+            while (!$streamInterface->eof()) {
+                echo $streamInterface->read(self::BUFFER_SIZE);
+            }
+            $streamInterface->close();
+        });
+
+        if (null === $fileSize = $streamInterface->getSize()) {
+            return $response;
+        }
+        $response->headers->set('Content-Length', strval($fileSize));
+
+        if ($streamInterface->isSeekable()) {
+            $response->headers->set('Accept-Ranges', $request->isMethodSafe() ? 'bytes' : 'none');
+        }
+
+        $range = $request->headers->get('Range');
+
+        if ($range === null) {
+            return $response;
+        }
+
+        list($start, $end) = explode('-', substr($range, 6), 2) + [0];
+
+        $end = ('' === $end) ? $fileSize - 1 : (int) $end;
+
+        if ('' === $start) {
+            $start = $fileSize - $end;
+            $end = $fileSize - 1;
+        } else {
+            $start = (int) $start;
+        }
+
+        if ($start > $end) {
+            return $response;
+        }
+
+        if ($start < 0 || $end > $fileSize - 1) {
+            $response->setStatusCode(StreamedResponse::HTTP_REQUESTED_RANGE_NOT_SATISFIABLE);
+            $response->headers->set('Content-Range', sprintf('bytes */%s', $fileSize));
+        } elseif (0 !== $start || $end !== $fileSize - 1) {
```

Comment:
The same object can be used here to define a different if test. 
I would also decouple it to a proper if, and use early return in the previous if statement.

Reply:
`if ($streamRange->isStarted())`

The StreamRange class should specify that the isOutOfRange() check should always occur before isStarted is called. If isStarted is called whily isOutOfRange() is true, isStarted should throw an error.

Reply:
I didn't follow you on the isStarted function but on a rangeRequested function. From my perspective it closer to what it is really: b681a9909166b190d18745755b2624f615a4982c